zaino-fetch/src/chain/mempool.rs

```diff
@@ -119,6 +119,35 @@ impl Mempool {
         Ok(txids.clone())
     }
 
+    /// Returns the txids currently in the mempool, filtered by exclude_txids.
+    ///
+    /// NOTE: THIS impl is inefficient and should be refactored with the addition of the internal mempool.
+    pub async fn get_filtered_mempool_txids(
+        &self,
+        exclude_txids: Vec<String>,
+    ) -> Result<Vec<String>, MempoolError> {
+        let mempool_txids = self.txids.read().await.clone();
+
+        let mut txids_to_exclude: HashSet<String> = HashSet::new();
+        for exclude_txid in &exclude_txids {
```

Comment:
this pattern I think is more common so maybe better?
```suggestion
        for exclude_txid in exclude_txids.iter() {
```